src/include/optimizer/optimize_context.h

```diff
@@ -22,18 +22,20 @@
 namespace peloton {
 namespace optimizer {
 
+template <class Operator, class OperatorType, class OperatorExpr>
 class OptimizerMetadata;
 
+template <class Operator, class OperatorType, class OperatorExpr>
 class OptimizeContext {
  public:
-  OptimizeContext(OptimizerMetadata *metadata,
+  OptimizeContext(OptimizerMetadata<Operator,OperatorType,OperatorExpr> *metadata,
                   std::shared_ptr<PropertySet> required_prop,
                   double cost_upper_bound = std::numeric_limits<double>::max())
       : metadata(metadata),
         required_prop(required_prop),
         cost_upper_bound(cost_upper_bound) {}
 
-  OptimizerMetadata *metadata;
+  OptimizerMetadata<Operator,OperatorType,OperatorExpr> *metadata;
```

Comment:
Just for style: the rest of the code names class members as metadata_ or required_prop_.
In addition, is there a particular reason why these members are public?